org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java

```diff
@@ -256,15 +256,10 @@ public DebugAdapter(Consumer<Events.DebugEvent> consumer, IProviderContext conte
     private Responses.ResponseBody launch(Requests.LaunchArguments arguments) {
         this.cwd = arguments.cwd;
         String mainClass = arguments.startupClass;
-        String classpath;
-        try {
-            // to-do: move to ls
-            classpath = AdapterUtils.computeClassPath(arguments.projectName, mainClass);
-            classpath = classpath.replaceAll("\\\\", "/");
-        } catch (CoreException e) {
-            Logger.logException("Failed to resolve classpath.", e);
-            return new Responses.ErrorResponseBody(new Types.Message(3001, "Cannot launch jvm.", null));
-        }
+        String classpath = arguments.classpath;
+        
+        // why
+        classpath = classpath.replaceAll("\\\\", "/");
```

Comment:
@testforstephen said that without it debugging might not work but i can not repro. i tested locally that without this vm can also be launched